File: 0326-power-of-three/0326-power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        max_int = 2**31-1
        k = int(math.log(max_int, 3))
        largest_p = 3 ** k
        return n > 0  and  (largest_p % n == 0) 

File: 0326-power-of-three/0326-power-of-three.java
class Solution {
    public boolean isPowerOfThree(int n) {
        if(n <=  0 ){
            return false;
        }
            int num = (int) Math.pow(2,31)-1;
            int k = (int) (Math.log(num)/ Math.log(3));
            long P = (long) Math.pow(3, k);
            return (P % n == 0);
                }
}